tests/trace/type_handlers/Content/content_test.py

```diff
@@ -0,0 +1,356 @@
+            assert saved_data == content.data
+
+            # Test save to directory (should use original filename)
+            content_with_filename = Content.from_path(image_file)
+            content_with_filename.save(tmpdir)
+            expected_path = Path(tmpdir) / image_file.name
+            assert expected_path.exists()
+
+    @pytest.mark.parametrize(
+        ("fixture_name", "extension", "mimetype"), MEDIA_TEST_PARAMS
+    )
+    def test_content_in_ops(
+        self, fixture_name: str, extension: str, mimetype: str, request
+    ):
+        """Test Content as input/output of weave ops."""
+        file_path = request.getfixturevalue(fixture_name)
+
+        # Op that returns Content object directly
+        @weave.op
+        def load_content_direct(path: str) -> Content:
+            return Content.from_path(path)
+
+        # Op that takes Content as input
+        @weave.op
+        def process_content(content: Content) -> dict:
+            return {
+                "size": content.size,
+                "extension": content.extension,
+                "mimetype": content.mimetype,
+            }
+
+        # Test direct Content return
+        content = load_content_direct(str(file_path))
+        assert isinstance(content, Content)
+        assert content.extension == extension
+
+        # Test Content as input
+        result = process_content(content)
+        assert result["size"] == content.size
+        assert result["extension"] == extension
+        assert result["mimetype"] == mimetype
+
+    def test_content_as_string(self):
+        """Test converting Content to string for text files."""
+        # Create a text content
+        text_data = "Hello, this is a test file!\nWith multiple lines."
+
+        content = Content.from_bytes(text_data.encode("utf-8"), extension="txt")
+        # The content class doesn't have an as_string method in the new API
+        # Instead we can decode the data directly
+        assert content.data.decode("utf-8") == text_data
+
+        # Test with different encoding
+        content_utf16 = Content.from_bytes(
+            text_data.encode("utf-16"), extension="txt", encoding="utf-16"
+        )
+        assert content_utf16.data.decode("utf-16") == text_data
+
+    def test_content_from_text(self):
+        """Test creating Content from text string."""
+        text_data = "Hello, this is a test file!\nWith multiple lines."
+
+        # Test Content.from_text()
+        content = Content.from_text(text_data, extension="txt")
+        assert content is not None
+        assert content.extension == ".txt"
+        assert content.mimetype == "text/plain"
+        assert content.data == text_data.encode("utf-8")
+        assert content.encoding == "utf-8"
+        assert content.input_type == "<class 'str'>"
+        assert content.content_type == "text"
+
+        # Test with custom encoding
+        content2 = Content.from_text(text_data, extension=".txt", encoding="utf-16")
```

Comment:
is there a difference between `.txt` and `txt`?

Reply:
Maybe worth some documentation in a README

Reply:
I'll document it. I changed it so that internally we use the conventional extension formatting which is to have a `.` leading. For the user they can omit or include.